### PR TITLE
More details

### PR DESCRIPTION
**Pull request by eazheng**

### CODE REVIEW

File: fbuapp/AppDelegate.m [hunk 1]
```diff
@@ -26,65 +26,65 @@ @implementation AppDelegate
...
     
     //create for persisting user
     if ([PFUser currentUser] != nil) {
         //if user is already logged in, take them to HomeViewController
-//        HomeViewController *homeViewController = [[HomeViewController alloc] init];
-//        homeViewController.modalPresentationStyle = UIModalPresentationFullScreen;
-//        homeViewController.modalTransitionStyle = UIModalTransitionStyleCoverVertical;
-//        UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:homeViewController];
-
+        //        HomeViewController *homeViewController = [[HomeViewController alloc] init];
```
Comment (belchercd) [inline fbuapp/AppDelegate.m]: update version to this with tab bar as the controller of the views

File: fbuapp/View Controllers/DetailsViewController.m [hunk 2]
```diff
@@ -37,10 +44,14 @@ typedef NS_ENUM(NSUInteger, SkillLevel) {
     ExpertSkill
 };
 
-@interface DetailsViewController () <UITableViewDelegate, UITableViewDataSource>
+@interface DetailsViewController () <UITableViewDelegate, UITableViewDataSource>//, PostCellDelegate>
```
Comment (belchercd) [inline fbuapp/View Controllers/DetailsViewController.m]: Do you need this comment?

File: fbuapp/View Controllers/DetailsViewController.m [hunk 3]
```diff
@@ -49,12 +60,19 @@ @implementation DetailsViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     
+    self.currentUser = @"Q78qpytNf5";//[PFUser currentUser].objectId;
```
Comment (belchercd) [inline fbuapp/View Controllers/DetailsViewController.m]: Switch

File: fbuapp/View Controllers/CreatePostViewController.m [hunk 4]
```diff
@@ -110,11 +113,45 @@ - (void)viewDidLoad {
     self.eventPriceField.leftView = priceImage;
     
     self.eventPriceField.delegate= self;
+    [self.eventPriceField setKeyboardType:UIKeyboardTypeNumberPad];
     
-    self.pillSelector = [[CategoryHeaderView alloc] initWithFrame:CGRectMake(0, self.eventCategoryLabel.frame.origin.y,self.scrollView.frame.size.width,60)];
-    
-    [self.contentView addSubview:self.pillSelector];
+    self.pillSelector = [[CategoryHeaderView alloc] initWithFrame:CGRectMake(0, 0, 0, 0)];
```
Comment (danthecodingman) [inline fbuapp/View Controllers/CreatePostViewController.m]: change this to the new initWithZero method once code is merged

File: fbuapp/View Controllers/DetailsViewController.m [hunk 5]
```diff
@@ -37,10 +44,14 @@ typedef NS_ENUM(NSUInteger, SkillLevel) {
     ExpertSkill
 };
 
-@interface DetailsViewController () <UITableViewDelegate, UITableViewDataSource>
+@interface DetailsViewController () <UITableViewDelegate, UITableViewDataSource>//, PostCellDelegate>
 
-@property (strong, nonatomic) IBOutlet UIView *overallView;
 @property (strong, nonatomic) IBOutlet UITableView *detailsTableView;
+@property UIBarButtonItem *navButton;
+@property NSString *currentUser;
```
Comment (danthecodingman) [inline fbuapp/View Controllers/DetailsViewController.m]: Is there a reason why these object properties are not (strong, nonatomic)?

File: fbuapp/View Controllers/DetailsViewController.m [hunk 6]
```diff
@@ -65,20 +83,92 @@ - (void)viewDidLoad {
...
-    button.backgroundColor = [UIColor blueColor];
+    NSArray *buttonColor = @[@0, @100, @255];
+    button.backgroundColor = [UIColor colorWithRGB:buttonColor];
+    [self.view addSubview:button];
+    
 
-    [self.overallView addSubview:button];
+    self.navButton = [[UIBarButtonItem alloc]init];
+    if ([self.currentUser isEqualToString:self.postAuthor.objectId]) {
+        NSLog(@"This is my post");
```
Comment (danthecodingman) [inline fbuapp/View Controllers/DetailsViewController.m]: Perhaps we should remove these log statements

File: fbuapp/View Controllers/DetailsViewController.m [hunk 7]
```diff
@@ -65,20 +83,92 @@ - (void)viewDidLoad {
...
+    }
+    
+    self.navButton.target = self;
+    // then we add the button to the navigation bar
+    self.navigationItem.rightBarButtonItem = self.navButton;
 }
 
+- (IBAction)onTapFavorited:(id)sender {
+    if (self.isFavorited == NO){
+        NSLog(@"favoriting post");
```
Comment (danthecodingman) [inline fbuapp/View Controllers/DetailsViewController.m]: These NSLogs as well

File: fbuapp/View Controllers/DetailsViewController.m [hunk 8]
```diff
@@ -110,35 +200,54 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
             LocationCell *cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier forIndexPath:indexPath];
             cell.locationNameLabel.text = self.post.eventLocationName;
             
-            PFGeoPoint *point = self.post.eventLocation;
-            //TODO: get distance away from current user.
-            NSString *dist = @"1.9";
+            NSString *dist = [PFGeoPoint distanceToPoint: self.post.eventLocation fromLocation: self.currentLocation];
             cell.locationDistanceLabel.text = [NSString stringWithFormat:@"About %@ miles away", dist];
```
Comment (danthecodingman) [inline fbuapp/View Controllers/DetailsViewController.m:202]: Could we make this one line with and never create a dist object?

File: fbuapp/View Controllers/DetailsViewController.m [hunk 9]
```diff
@@ -110,35 +200,54 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
+            cell.authorNameLabel.text = [NSString stringWithFormat:@"%@ %@", self.postAuthor[@"firstName"], self.postAuthor[@"lastName"]];
+            cell.usernameLabel.text = self.postAuthor[@"username"];
+            
+            cell.fbProfileView.profileID = self.postAuthor[@"fbUserId"];
             
-            if (self.post.authorRole == 1 && ![self.post.eventPrice isEqual:@0]) { //author wants to teach
+            if (self.post.authorRole != LearnRole && ![self.post.eventPrice isEqual:@0]) { //author wants to teach
                 cell.authorPriceLabel.text = [NSString stringWithFormat:@"Your cost: $%@", self.post.eventPrice];
+                NSArray *redColor = @[@180, @0, @0];
+                cell.authorPriceLabel.textColor = [UIColor colorWithRGB: redColor];
```
Comment (danthecodingman) [inline fbuapp/View Controllers/DetailsViewController.m:216]: Make all these color settings one line

**Pull request merged**